src/Servers/Kestrel/Transport.MsQuic/Internal/SocketNativeMethods.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal
+{
+    internal class SocketNativeMethods
```

Comment:
I'll add a linux version of this method in the next PR (when I can actually do testing on linux).

Reply:
Actually, I'm just going to remove it as I think there should be a method in corefx that does this conversion for us.